src/lib/crypto/bn.c

```diff
@@ -205,32 +206,23 @@ PGPV_BN_clear_free(PGPV_BIGNUM *a)
     PGPV_BN_free(a);
 }
 
-int
-PGPV_BN_num_bytes(const PGPV_BIGNUM *a)
+bool
+BN_num_bits(const PGPV_BIGNUM *a, size_t *bits)
 {
-    size_t num_bytes;
-    if (a == NULL) {
-        return -1;
+    if (!a || botan_mp_num_bits(a->mp, bits)) {
+        return false;
     }
-
-    if (botan_mp_num_bytes(a->mp, &num_bytes) < 0) {
-        return -1;
-    }
-    return num_bytes;
+    return true;
 }
 
-int
-PGPV_BN_num_bits(const PGPV_BIGNUM *a)
+bool
+BN_num_bytes(const PGPV_BIGNUM *a, size_t *bits)
 {
-    size_t num_bits;
-    if (a == NULL) {
-        return -1;
-    }
-
-    if (botan_mp_num_bits(a->mp, &num_bits) < 0) {
-        return -1;
+    if (BN_num_bits(a, bits)) {
```

Comment:
I don't think there is big difference. One argument is - because then I can handle errors from botan in single place. I guess one is as good as the other one internally botan_mp_num_bits does same thing as bn_mp_num_bytes.